fe1-web/src/features/notification/screens/NotificationScreen.tsx

```diff
@@ -0,0 +1,153 @@
+import { NotificationStackParamList } from '../navigation/NotificationStackParamList';
+import { discardNotifications, NotificationState, makeAllNotificationsSelector } from '../reducer';
+
+interface ListSeparatorItem {
+  title: string;
+}
+
+interface NotificationItem extends NotificationState {
+  isLastItem: boolean;
+}
+
+const NotificationScreenStyles = StyleSheet.create({
+  notificationItem: {
+    paddingTop: 16,
+    paddingBottom: 16,
+    borderBottomColor: '#000',
+    borderBottomWidth: 1,
+  } as ViewStyle,
+
+  lastNotificationItem: {
+    borderBottomWidth: 0,
+  } as ViewStyle,
+});
+
+type NavigationProps = StackScreenProps<
+  NotificationStackParamList,
+  'NotificationNavigation Notifications'
+>;
+
+const NotificationScreen = () => {
+  const navigation = useNavigation<NavigationProps['navigation']>();
+
+  const laoId = NotificationHooks.useCurrentLaoId();
+  const selectAllNotifications = useMemo(
+    () => makeAllNotificationsSelector(laoId.valueOf()),
+    [laoId],
+  );
+  const notifications = useSelector(selectAllNotifications);
+  const notificationData = useMemo(() => {
+    const read: NotificationItem[] = [];
+    const unread: NotificationItem[] = [];
+    for (const notification of notifications) {
+      if (notification.hasBeenRead) {
+        read.push({ ...notification, isLastItem: false });
+      } else {
+        unread.push({ ...notification, isLastItem: false });
+      }
+    }
+
+    // sort in descending order, i.e. newest/latest first
+    read.sort((a, b) => b.timestamp - a.timestamp);
+    unread.sort((a, b) => b.timestamp - a.timestamp);
+
+    const items: (ListSeparatorItem | NotificationItem)[] = [];
+    if (unread.length > 0) {
+      // set isLastItem properties
+      unread[unread.length - 1].isLastItem = true;
+
+      items.push({ title: 'Notifications' } as ListSeparatorItem, ...unread);
+    }
+
+    if (read.length > 0) {
+      // set isLastItem properties
+      read[read.length - 1].isLastItem = true;
+
+      items.push({ title: 'Read Notifications' } as ListSeparatorItem, ...read);
+    }
+
+    return items;
+  }, [notifications]);
+
+  const notificationTypes = NotificationHooks.useNotificationTypes();
+
+  const onClearNotifications = () => {
+    // call custom delete function on all notifications
+    for (const notification of notifications) {
+      const deleteFn = notificationTypes.find((t) => t.isOfType(notification))?.delete;
+
+      // if a delete function was provided for this type, then call it
+      if (deleteFn) {
+        deleteFn(notification);
+      }
+    }
+    // remove notifications from the notification reducer
+    dispatch(
+      discardNotifications({
+        laoId: laoId.valueOf(),
+        notificationIds: notifications.map((n) => n.id),
+      }),
+    );
+  };
+
+  return (
+    <ScreenWrapper>
+      <FlatList
+        data={notificationData}
+        keyExtractor={(item) => ('id' in item ? item.id.toString() : item.title)}
```

Comment:
Be careful with that. If 2 items happen to have the same id, the application will crash immediately.

Reply:
Yeah that's what I was trying to point out, thanks @pierluca . The id is fine, but the title could be the same if the id happens to be undefined.

Reply:
I added comments and a check at the end of the `useMemo` that checks for uniqueness and throws an error otherwise. Do you think this suffices or did you have anything else in mind? I think this is easier than adding a complex mechanism for generating unique identifiers for the headings?